api_tests/files/views/test_file_list.py

```diff
@@ -1,138 +1,159 @@
+
+    @pytest.fixture()
+    def node(self, user):
+        return ProjectFactory(creator=user)
+
+    @pytest.fixture()
+    def file(self, user, node):
+        return api_utils.create_test_file(
+            node, user, filename='file_one')
+
+    @pytest.fixture()
+    def deleted_file(self, user, node):
+        deleted_file = api_utils.create_test_file(
+            node, user, filename='file_two')
+        deleted_file.delete(user=user, save=True)
+        return deleted_file
+
+    def test_does_not_return_trashed_files(self, app, user, node, file, deleted_file):
+        res = app.get(
+            '/{}nodes/{}/files/osfstorage/'.format(API_BASE, node._id),
+            auth=user.auth
         )
         data = res.json.get('data')
-        nt.assert_equal(len(data), 1)
-
-class TestFileFiltering(ApiTestCase):
-    def setUp(self):
-        super(TestFileFiltering, self).setUp()
-        self.user = AuthUserFactory()
-        self.node = ProjectFactory(creator=self.user)
-        self.file1 = api_utils.create_test_file(
-            self.node, self.user, filename='file1')
-        self.file2 = api_utils.create_test_file(
-            self.node, self.user, filename='file2')
-        self.file3 = api_utils.create_test_file(
-            self.node, self.user, filename='file3')
-        self.file4 = api_utils.create_test_file(
-            self.node, self.user, filename='file4')
-
-    def test_get_all_files(self):
-        res = self.app.get(
-            '/{}nodes/{}/files/osfstorage/'.format(API_BASE, self.node._id),
-            auth=self.user.auth
+        assert len(data) == 1
+
+@pytest.mark.django_db
+class TestFileFiltering:
+
+    @pytest.fixture()
+    def node(self, user):
+        return ProjectFactory(creator=user)
+
+    @pytest.fixture()
+    def file_one(self, user, node):
+        return api_utils.create_test_file(
+            node, user, filename='file_one')
+
+    @pytest.fixture()
+    def file_two(self, user, node):
+        return api_utils.create_test_file(
+            node, user, filename='file_two')
+
+    @pytest.fixture()
+    def file_three(self, user, node):
+        return api_utils.create_test_file(
+            node, user, filename='file_three')
+
+    @pytest.fixture()
+    def file_four(self, user, node):
+        return api_utils.create_test_file(
+            node, user, filename='file_four')
+
+    def test_get_all_files(self, app, user, node, file_one, file_two, file_three, file_four):
+        res = app.get(
+            '/{}nodes/{}/files/osfstorage/'.format(API_BASE, node._id),
+            auth=user.auth
         )
         data = res.json.get('data')
-        nt.assert_equal(len(data), 4)
+        assert len(data) == 4
 
-    def test_filter_on_tag(self):
-        self.file1.add_tag('new', Auth(self.user))
-        self.file2.add_tag('new', Auth(self.user))
-        res = self.app.get(
+    def test_filter_on_single_tag(self, app, user, node, file_one, file_two, file_three, file_four):
```

Comment:
Also for those, I am particularly careful about not adding/removing multiple times but using multiple individual files for safety.